signal/reporter.go

```diff
@@ -29,59 +29,61 @@ type Reporter interface {
 }
 
 func PullReport(r Reporter, c config.Config) error {
-	log.Debugf("Attempting to pull report from %s", r.GetURL())
-	url, err := url.Parse(r.GetURL())
-	if err != nil {
-		return err
-	}
+	for _, u := range r.GetURLs() {
+		log.Debugf("Attempting to pull report from %s", u)
+		url, err := url.Parse(u)
+		if err != nil {
+			return err
+		}
 
-	client := http.Client{
-		Timeout: time.Duration(5 * time.Second),
-	}
+		client := http.Client{
+			Timeout: time.Duration(5 * time.Second),
+		}
 
-	if url.Scheme == "https" {
-		client.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		if url.Scheme == "https" {
+			client.Transport = &http.Transport{
```

Comment:
looks like we might make multiple calls to Mesos? in that case it might be nice to leverage connection pooling (Transport reuse); can set `client.Transport` to `nil` when scheme != https, otherwise reuse the same Transport instance for https connections?